### PR TITLE
Initial Implementation/user mangement

### PR DESCRIPTION
**Pull request by two2tee**
Implememented TeamHandler
Implememented UserHandler
Implememented UserFacade
Implememented TeamFacade
Implememented Validators
Implememented UserManagementProfile for automapper
Implememented tests
Added Documentation

~~BUG:~~
~~Automapper is unable to map TeamStorage and UserStorage probably. ~~
~~It is throwing exceptions of incorrect mapping. I'll bring this up for next meeting.~~

EDIT
Fixed automapper bug! All is working now. All tests pass! Ready to go!

### CODE REVIEW

**Comment by two2tee** [comment]
Fixed bug related to automapper!
Ready for deployment!
All test pass!

File: ApplicationLogicTests/UserManagement/TeamFacadeTests.cs [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+        }
+
+        /// <summary>
+        /// Test if read with parameters returns correct numbers of teams
+        /// </summary>
+        [TestMethod]
+        public void GetAllTeams_Valid_ReturnsCorrectNumberOfTeams_Test()
+        {
+            //Arrange
+            var team1 = new StoredTeam {Name = "name1", MetaData = "metaData1"};
```
Comment (TVAO) [inline ApplicationLogicTests/UserManagement/TeamFacadeTests.cs]: Could use TestInitialize to make same list of users for all tests and avoid code duplication.

File: ApplicationLogicTests/UserManagement/TeamFacadeTests.cs [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+            //Arrange
+            var idToRead = 0;
+            _repositoryMock.Setup(r => r.Read(idToRead)).Returns(_storedTeam);
+            var teamFacade = new TeamFacade(_repositoryMock.Object);
+
+            //Act
+            var returnedTeam = teamFacade.Read(idToRead);
+
+            //Assert
+            Assert.IsTrue(_team.Name == returnedTeam.Name);
```
Comment (TVAO) [inline ApplicationLogicTests/UserManagement/TeamFacadeTests.cs]: Only one assert

File: ApplicationLogicTests/UserManagement/UserTest.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    /// This class test the user in application logic.
+    /// </summary>
+    [TestClass]
+    public class UserTest
+    {
+        /// <summary>
+        /// Creates a valid user.
+        /// </summary>
+        [TestMethod]
+        public void CreateUser_NewUser_Valid_Test()
```
Comment (TVAO) [inline ApplicationLogicTests/UserManagement/UserTest.cs]: Very nice textual description and structure

File: ApplicationLogics/StorageFasade/TeamFacade.cs [hunk 4]
```diff
@@ -1,38 +1,87 @@
...
+    /// This class will handle Teams and convert them the the propriate object that are to be propagated
+    /// </summary>
     public class TeamFacade : IFacade<Team>
     {
-        public int Create(Team item)
+        private readonly IRepository<StoredTeam> _teamRepository;
+
+        public TeamFacade(IRepository<StoredTeam> repository)
         {
-            throw new NotImplementedException();
```
Comment (TVAO) [inline ApplicationLogics/StorageFasade/TeamFacade.cs]: Should take in Users and not StoredUsers. Only when calling repository_create the underlying storage repo should be called?

**Pull request merged**